### PR TITLE
Add expose strategy proposal based on HTTP tunneling and SNI

### PR DESCRIPTION
**Pull request by irozzo-1A**
**What this PR does / why we need it**:
This proposal addresses issue #4942

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: docs/proposals/http-tunnel-expose-strategy.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
+# **Exposing control-plane services using SNI and HTTP tunneling**
+
+Author: Youssef Azrak (@youssefazrak), Iacopo Rozzo (@irozzo-1A)
+
+Status: Draft proposal.
+
+
+## **Goals**
```
Comment (xrstf) [inline docs/proposals/http-tunnel-expose-strategy.md]: No need to have this text be a heading AND bold.

File: docs/proposals/http-tunnel-expose-strategy.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+*   Reduce cost by using only one Load Balancer.
+*   AWS default quota: 50 Load Balancer per region and 50 listener per Load Balancer.
+
+
+There are two main points we want to address with this proposal:
+
+*   Reduce cost by using only one Load Balancer.
+*   AWS default quota: 50 Load Balancer per region and 50 listener per Load Balancer.
+
```
Comment (xrstf) [inline docs/proposals/http-tunnel-expose-strategy.md:21]: This is duplicated.

File: docs/proposals/http-tunnel-expose-strategy.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+
+<div style="text-align:center">
+  <img src="images/http-tunnel-expose-strategy-sni.png" width="100%" />
+</div>
+
+
+### Accessing the Kube Apiserver through the Kubernetes service
+
+PODs that need to communicate with the Apiserver normally rely on the KUBERNETES_SERVICE environment variable that contains the cluster IP of the kubernetes service in the default namespace. As we already anticipated in this scenario, we cannot rely on SNI as the client will use the IP directly. In order to bypass this limitation, the TCP traffic is encapsulated in an HTTP tunnel based on CONNECT method, that can be routed based on the host header ([HTTP1.x][http_upgrade_tls_rfc]) or the authority header ([HTTP2][http2_connect_rfc]).
```
Comment (xrstf) [inline docs/proposals/http-tunnel-expose-strategy.md]: What's a P.O.D.?

**Review by xrstf — CHANGES_REQUESTED** [review]
minor things

File: docs/proposals/http-tunnel-expose-strategy.md (hunk 1 above)
Comment (youssefazrak) [reply docs/proposals/http-tunnel-expose-strategy.md]: Fixed

File: docs/proposals/http-tunnel-expose-strategy.md (hunk 2 above)
Comment (youssefazrak) [reply docs/proposals/http-tunnel-expose-strategy.md:21]: Fixed

File: docs/proposals/http-tunnel-expose-strategy.md (hunk 3 above)
Comment (youssefazrak) [reply docs/proposals/http-tunnel-expose-strategy.md]: Fixed

**Review by xrstf — APPROVED** [review]
/approve

**Pull request merged**